test/many-streams.test.js

```diff
@@ -131,40 +131,39 @@ async function makeTableP({gbqClient, gbqDatasetName, gbqTableName}) {
 }
 
 function createReadStream() {
+  var results = Array(streamLength).fill(stringifyObject(testData))
 
   return new Readable({
-
-    objectMode   : true,
-    highWaterMark: streamLength,
-
     read(_size) {
-
       const thisStream = this
+      this.push(results.pop())
```

Comment:
Instead of:

- For every pull from this stream, emit 200 rows at once

I changed it to:

- For every pull from this stream, emit 1 row

Then, you don't have to get into the flow control aspect of Node stream internals with `highWatermark`-- we will let the requesting stream pull as it needs and get one thing at a time.

Reply:
That is way better thank you.

Obviously it's not important in this case, but do you have any idea if reading one item at a time would have performance implications in other projects?

Reply:
I can't say definitively, but in general, I would trust Node's internals to handle as much as possible. So switching to the pattern of pull (1) -> get (1) where possible *should* show performance improvements. If you aren't seeing good performance, play with `highWaterMark` (16kb default / 16 objects default for objectMode), but still return one thing at a time from `read`.

(If you find that to be bad advice, let me know!)

Reply:
Thank you, this sounds like good advice and I'll write readable streams this way in the future.